src/app/shared/search/search-result.component.ts

```diff
@@ -89,12 +91,44 @@ export class SearchResultComponent implements OnChanges {
       this.resultView.columns = this.resultView.columns.sort((a: SearchResultViewColumn, b: SearchResultViewColumn) => {
         return a.order - b.order;
       });
+
+      this.hydrateResultView();
     }
     if (changes['page']) {
       this.selected.page = (changes['page']).currentValue;
     }
   }
 
+  /**
+   * Hydrates result view with case field definitions.
+   */
+  // A longer term resolution is to move this piece of logic to the backend
+  hydrateResultView(): void {
```

Comment:
it took me a while to figure out what's going on here. I think this new code could use some refactoring and better naming to make it more self explanatory. 
Also, I think instead of adding new fields to SearchResultViewItem (hydrated_case_fields and columns), we could consider introducing a new concept/type in our domain, for example DisplayedSearchItem. And we 
could have SearchResultComponent create list of DisplayedSearchItem from the SearchResultViewItem.
The SearchResultComponent view could then operate on the DisplayedSearchItem only.
This way we keep SearchResultViewItem being a one to one mapping with the data returned from the backend api.

Reply:
Definitely moving that logic to a service is a must. Everything you mentioned @mario-paniccia is good I think; though for me the proper fix would be to move the whole label substitution for read-only views to the backend. This would remove the need for most of the code introduced by this PR and try and prevent the complexity of the FE from increasing.

Reply:
@vlaurin that's true yes. And I think we should do both. Because my suggestion is not strictly related to label substitution. But it's more an improvement to the design of the search component in general